src/pages/ChattingRoom.tsx

```diff
@@ -0,0 +1,209 @@
+import { useState, useEffect, useRef } from "react";
+import { useRecoilState, useRecoilValue } from "recoil";
+import { currentUsersState } from "../recoil/recoil";
+import styled from "styled-components";
+import { chatInterface, userInterface } from "../interfaces/interface";
+import usersData from "../json/usersData.json";
+import chatsData from "../json/chatsData.json";
+import Alert from "../components/Alert";
+import SmallButton from "../components/SmallButton";
+import Profile from "../components/Profile";
+import LeftChat from "../components/LeftChat";
+import RightChat from "../components/RightChat";
+
+const ChattingRoom = () => {
+  const currentUsers = useRecoilValue<userInterface[]>(currentUsersState);
+  const [typingUser, setTypingUser] = useState<userInterface>(
+    usersData.users[0]
+  );
+  const nonTypingUser = currentUsers.filter(
+    (user) => user.userId !== typingUser.userId
+  )[0];
+  const [chatList, setChatList] = useState<chatInterface[]>(
+    chatsData.chattings[0].chatList
+  ); // TODO - recoil로 변경
+  // const [chatList, setChatList] = useRecoilState(currentChatListState);
+
+  const [inputText, setInputText] = useState<string>("");
+  const isTyping: boolean = inputText.trim() ? true : false;
+
+  const [isVisibleAlert, setIsVisibleAlert] = useState<boolean>(false);
+
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
+  const bottomDivRef = useRef<HTMLDivElement>(null);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setInputText(e.target.value);
+  };
+
+  const handleEnterKeyUp = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
+    if (!e.shiftKey && e.key === "Enter") {
+      handleSubmit();
+    }
+  };
+
+  const handleSubmit = (e?: React.FormEvent<HTMLFormElement>) => {
+    e?.preventDefault();
+    if (inputText.trim()) {
+      setChatList([
+        ...chatList,
+        {
+          userId: typingUser.userId,
+          message: inputText,
+          date: String(new Date()),
+          chatId: `ex${chatList.length}`,
+        },
+      ]);
+      setInputText("");
+      textareaRef.current?.focus();
+    } else {
+      setInputText("");
+      alert("메세지를 입력해주세요 :)");
+    }
```

Comment:
input이 빈칸인 경우 submit을 하면 alert호출이 되고 다시 focus가 되게 하면서 코드의 중복을 피하려면 이렇게 작성해도 좋을 것 같습니다!
```suggestion
    if (inputText.trim()) {
      setChatList([
        ...chatList,
        {
          userId: typingUser.userId,
          message: inputText,
          date: String(new Date()),
          chatId: `ex${chatList.length}`,
        },
      ]);
    } else {
      alert("메세지를 입력해주세요 :)");
    }
    setInputText('');
    textareaRef.current?.focus();
```